modules/caddyhttp/reverseproxy/selectionpolicies.go

```diff
@@ -110,8 +110,8 @@ func (r *WeightedRoundRobinSelection) UnmarshalCaddyfile(d *caddyfile.Dispenser)
 		if err != nil {
 			return d.Errf("invalid weight value '%s': %v", weight, err)
 		}
-		if weightInt < 1 {
-			return d.Errf("invalid weight value '%s': weight should be non-zero and positive", weight)
+		if weightInt < 0 {
+			return d.Errf("invalid weight value '%s': weight should be positive", weight)
```

Comment:
```suggestion
			return d.Errf("invalid weight value '%s': weight should be non-negative", weight)
```

0 isn't positive but is also allowed now :wink:

Reply:
Good question, I don't have the enough knowledge to write the test case for this. Any guide would be helpful.

Reply:
Have you at least tested various configs yourself to ensure they work? If so that's probably enough to start with.

But it looks like the existing tests are in `selectionpolicies_test.go` at `func TestWeightedRoundRobinPolicy(t *testing.T) {`. You could test a zero-weight in there as well.

Reply:
Thank you for your reply, will work on generating the test case.

Reply:
@peanutduck this still needs tests. Are you interested in finishing this up?

Reply:
Hi @francislavoie 

I try to add test case to this PR here https://github.com/peanutduck/caddy/pull/1